flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/MatchRecognizeTestPrograms.java

```diff
@@ -0,0 +1,300 @@
+
+/** {@link TableTestProgram} definitions for testing {@link StreamExecMatch}. */
+public class MatchRecognizeTestPrograms {
+    static final Row[] SIMPLE_DATA = {
+        Row.of(1L, "a"),
+        Row.of(2L, "z"),
+        Row.of(3L, "b"),
+        Row.of(4L, "c"),
+        Row.of(5L, "d"),
+        Row.of(6L, "a"),
+        Row.of(7L, "b"),
+        Row.of(8L, "c"),
+        Row.of(9L, "a"),
+        Row.of(10L, "b")
+    };
+
+    static final Row[] SIMPLE_DATA2 = {Row.of(11L, "c")};
+
+    static final Row[] COMPLEX_DATA = {
+        Row.of("ACME", 1L, 19, 1),
+        Row.of("BETA", 2L, 18, 1),
+        Row.of("ACME", 3L, 17, 2),
+        Row.of("ACME", 4L, 13, 3),
+        Row.of("BETA", 5L, 16, 2),
+        Row.of("ACME", 6L, 20, 4)
+    };
+
+    static final Row[] COMPLEX_DATA2 = {Row.of("BETA", 7L, 22, 4)};
+
+    static final TableTestProgram SIMPLE_MATCH =
+            TableTestProgram.of("simple-match", "simple match recognize test")
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("MyTable")
+                                    .addSchema(
+                                            "id bigint", "name varchar", "proctime as PROCTIME()")
+                                    .producedBeforeRestore(SIMPLE_DATA)
+                                    .producedAfterRestore(SIMPLE_DATA2)
+                                    .build())
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("MySink")
+                                    .addSchema("a bigint", "b bigint", "c bigint")
+                                    .consumedBeforeRestore(Row.of(6L, 7L, 8L))
+                                    .consumedAfterRestore(Row.of(9L, 10L, 11L))
+                                    .build())
+                    .runSql(
+                            "insert into MySink"
+                                    + " SELECT T.aid, T.bid, T.cid\n"
+                                    + "     FROM MyTable MATCH_RECOGNIZE (\n"
+                                    + "             ORDER BY proctime\n"
+                                    + "             MEASURES\n"
+                                    + "             `A\"`.id AS aid,\n"
+                                    + "             \u006C.id AS bid,\n"
+                                    + "             C.id AS cid\n"
+                                    + "             PATTERN (`A\"` \u006C C)\n"
+                                    + "             DEFINE\n"
+                                    + "                 `A\"` AS name = 'a',\n"
+                                    + "                 \u006C AS name = 'b',\n"
+                                    + "                 C AS name = 'c'\n"
+                                    + "     ) AS T")
+                    .build();
+
+    static final TableTestProgram COMPLEX_MATCH =
+            TableTestProgram.of("complex-match", "complex match recognize test")
```

Comment:
Could you please adapt the id according to the javadoc:
https://github.com/apache/flink/blob/52d8d3583e5c989da84126a8805ab335408c46c2/flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/test/program/TableTestProgram.java#L117
?

Reply:
Ah, yes; didn't see that.  I was reusing identifiers from the existing tests!

Updated.